core/domain/skill_services.py

```diff
@@ -230,6 +232,239 @@ def get_all_skill_summaries():
+            total_count: number. Total count of skill summaries.
+    """
+    skill_summaries, next_cursor, more, total_count = (
+        get_skill_summaries_in_batches(
+            num_queries_to_fetch, cursor=cursor, sort_by=sort_by))
+    augmented_skill_summaries = [skill_domain.AugmentedSkillSummary(
+        summary.id,
+        summary.description,
+        summary.language_code,
+        summary.version,
+        summary.misconception_count,
+        summary.worked_examples_count,
+        summary.skill_model_created_on,
+        summary.skill_model_last_updated) for summary in skill_summaries]
+
+    assigned_skill_ids = (
+        topic_services.
+        get_all_skill_ids_assigned_to_some_topic_with_topic_details())
+
+    for augmented_skill_summary in augmented_skill_summaries:
+        skill_summary_id = augmented_skill_summary.id
+        if skill_summary_id in assigned_skill_ids:
+            augmented_skill_summary.update_topic_name(
+                assigned_skill_ids[skill_summary_id]['topic_name'])
+            augmented_skill_summary.update_classroom_name(
+                assigned_skill_ids[skill_summary_id]['classroom_name'])
+
+    return augmented_skill_summaries, next_cursor, more, total_count
+
+
+def filter_skills_by_status(augmented_skill_summaries, status):
+    """Returns the skill summary dicts after filtering by status.
+
+    Args:
+        augmented_skill_summaries: list(AugmentedSkillSummary). The list
+        of augmented skill summaries.
+        status: str. The status of the skill.
+
+    Returns:
+        list(AugmentedSkillSummary). The list of AugmentedSkillSummaries
+        matching the given status.
+    """
+
+    if not status or status == constants.SKILL_STATUS_OPTIONS['ALL']:
+        return augmented_skill_summaries
+
+    elif status == constants.SKILL_STATUS_OPTIONS['UNASSIGNED']:
+        unassigned_augmented_skill_summaries = []
+        for augmented_skill_summary in augmented_skill_summaries:
+            if not augmented_skill_summary.topic_name:
+                unassigned_augmented_skill_summaries.append(
+                    augmented_skill_summary)
+
+        return unassigned_augmented_skill_summaries
+
+    elif status == constants.SKILL_STATUS_OPTIONS['ASSIGNED']:
+        assigned_augmented_skill_summaries = []
+        for augmented_skill_summary in augmented_skill_summaries:
+            if augmented_skill_summary.topic_name:
+                assigned_augmented_skill_summaries.append(
+                    augmented_skill_summary)
+
+        return assigned_augmented_skill_summaries
+
+
+def filter_skills_by_classroom(augmented_skill_summaries, classroom_name):
+    """Returns the skill summary dicts after filtering by classroom_name.
+
+    Args:
+        augmented_skill_summaries: list(AugmentedSkillSummary).
+            The list of augmented skill summaries.
+        classroom_name: string. The name of the classroom.
```

Comment:
str, right? (Or None? Add an explanation for what that means, too.)

Reply:
Done.